File: code/drasil-code/test/golden/cpp/FileTests/FileTests.cpp
#include <algorithm>
#include <cassert>
#include <fstream>
#include <iostream>
#include <limits>
#include <string>
#include <vector>

using std::ifstream;
using std::ofstream;
using std::string;
using std::vector;

int main(int argc, const char *argv[]) {
    ofstream fileToWrite;
    fileToWrite.open("testText.txt", std::fstream::out);
    fileToWrite << 0;
    fileToWrite << 0.89 << std::endl;
    fileToWrite << "ello";
    fileToWrite << "bye" << std::endl;
    fileToWrite << "!!" << std::endl;
    fileToWrite.close();
    ifstream fileToRead;
    fileToRead.open("testText.txt", std::fstream::in);
    string fileLine;
    fileToRead >> fileLine;
    fileToRead.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    fileToRead.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    assert(fileLine != "" && "First line should not be empty.");
    vector<string> fileContents(0);
    
    string nextLine;
    while (std::getline(fileToRead, nextLine)) {
        fileContents.push_back(nextLine);
    }
    
    std::cout << "[";
    for (int list_i1 = 0; list_i1 < (int)(fileContents.size()) - 1; list_i1++) {
        std::cout << fileContents.at(list_i1);
        std::cout << ", ";
    }
    if ((int)(fileContents.size()) > 0) {
        std::cout << fileContents.at((int)(fileContents.size()) - 1);
    }
    std::cout << "]" << std::endl;
    assert((int)(fileContents.size()) > 0 && "fileContents should not be empty.");
    fileToRead.close();
    
    return 0;
}

File: code/drasil-code/test/golden/csharp/FileTests/FileTests.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;

public class FileTests {
    
    public static void Main(string[] args) {
        StreamWriter fileToWrite;
        fileToWrite = new StreamWriter("testText.txt", false);
        fileToWrite.Write(0);
        fileToWrite.WriteLine(0.89);
        fileToWrite.Write("ello");
        fileToWrite.WriteLine("bye");
        fileToWrite.WriteLine("!!");
        fileToWrite.Close();
        StreamReader fileToRead;
        fileToRead = new StreamReader("testText.txt");
        string fileLine;
        fileLine = fileToRead.ReadLine();
        fileToRead.ReadLine();
        Debug.Assert( fileLine != "" , "First line should not be empty.");
        List<string> fileContents = new List<string>(0);
        
        while (!(fileToRead.EndOfStream)) {
            fileContents.Add(fileToRead.ReadLine());
        }
        
        Console.Write("[");
        for (int list_i1 = 0; list_i1 < fileContents.Count - 1; list_i1++) {
            Console.Write(fileContents[list_i1]);
            Console.Write(", ");
        }
        if (fileContents.Count > 0) {
            Console.Write(fileContents[fileContents.Count - 1]);
        }
        Console.WriteLine("]");
        Debug.Assert( fileContents.Count > 0 , "fileContents should not be empty.");
        fileToRead.Close();
    }
}
